src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -1748,10 +1764,13 @@ public SyntheticPlugin header(String key, String value) {
          * @return back to the rule builder
          */
         public RealJenkinsRule done() {
-            return RealJenkinsRule.this;
+            if (rjr == null) {
+                throw new IllegalStateException("May only be called after RealJenkinsRule.addSyntheticPlugin");
+            }
+            return rjr;
         }
 
-        void writeTo(File jpi, String defaultJenkinsVersion) throws IOException, URISyntaxException {
+        public void writeTo(File jpi, String defaultJenkinsVersion) throws IOException, URISyntaxException {
```

Comment:
Better introduce Javadoc especially for the nonobvious `defaultJenkinsVersion` param (which we should likely rename to just `jenkinsVersion`).

Reply:
[056e148](https://github.com/jenkinsci/jenkins-test-harness/pull/920/commits/056e148972f94c931cf409cfceb5bba1a9d77c03)